src/Npgsql/NpgsqlConnector.cs

```diff
@@ -210,6 +210,8 @@ internal void FlagAsWritableForMultiplexing()
 
         bool _sendResetOnClose;
 
+        NpgsqlException? originalTimeoutException = null;
```

Comment:
```suggestion
        NpgsqlException? _originalTimeoutException;
```

Reply:
More importantly, this looks like it could be a local variable inside ReadMessageLong.

Reply:
That's what I also thought at first. I was testing with a `pg_sleep` function, and in this case we fail with a timeout on reading of a `ParseCompleteMessage` here:
https://github.com/npgsql/npgsql/blob/2573a48c4612b2217d502bc35f1c6d37c175f95c/src/Npgsql/NpgsqlDataReader.cs#L412

So if the `_originalTimeoutException` is a local variable, it would get lost between reads.

Reply:
I see - this is so that:
* We get a timeout while waiting for one message
* We send a cancel request
* We get the actual message and return it, processing continues
* Then at some later point we get the error response for the cancel request, and want to throw the original timeout exception from before.

Makes sense.

Reply:
Yep, exactly!